ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/LabelClickSpikesTest.java

```diff
@@ -1,45 +1,56 @@
 package com.mozilla.telemetry.contextualservices;
 
-import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Iterables;
-import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
 import java.util.Arrays;
 import java.util.stream.StreamSupport;
-import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
-import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessageWithAttributesCoder;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.SchemaCoder;
 import org.apache.beam.sdk.testing.PAssert;
 import org.apache.beam.sdk.testing.TestPipeline;
 import org.apache.beam.sdk.testing.TestStream;
 import org.apache.beam.sdk.testing.TestStream.Builder;
 import org.apache.beam.sdk.transforms.Values;
 import org.apache.beam.sdk.transforms.WithKeys;
 import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
 import org.joda.time.Duration;
 import org.junit.Rule;
 import org.junit.Test;
 
 public class LabelClickSpikesTest {
 
+  private SponsoredInteraction.Builder getTestInteraction() {
+    return SponsoredInteraction.builder().setInteractionType("click").setSource("topsite")
+        .setFormFactor("phone").setContextId("1");
+  }
+
   @Rule
   public TestPipeline pipeline = TestPipeline.create();
 
+  private SchemaCoder<SponsoredInteraction> getCoder() {
```

Comment:
why not re-use the coder definition from `AggregateImpressions.java`?

Reply:
I moved the method to a static on `SponsoredInteraction`